Connection.cs

```diff
@@ -0,0 +1,125 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows;
+
+namespace Baigiamasis
+{
+    internal class Connection
+    {
+        public IPEndPoint EndPoint { get; set; }
+        public TcpClient Client { get; set; }
+        public NetworkStream Stream { get; set; }
+        public CancellationTokenSource Listening { get; set; }
+        private byte[] Buffer { get; set; }
+        public int BufferSize { get; set; } = 65536;
+
+        public User userInfo;
+        public List<User> userList;
+
+        public Connection(string iPAddress, int port, User info)
+        {
+            EndPoint = new IPEndPoint(IPAddress.Parse(iPAddress), port);
+            Client = new TcpClient();
+            Buffer = new byte[BufferSize];
+            userList = new List<User>();
+            userInfo = info;
+        }
+
+        public void Connect(Action<Message> T)
+        {
+            try
+            {
+                Client.Connect(EndPoint);
+                Stream = Client.GetStream();
+                Listening = new CancellationTokenSource();
+                Stream.Write(Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(userInfo)));
+                RX(T);
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(ex.Message);
+            }
+        }
+
+        public void Disconnect()
+        {
+            try
+            {
+                Listening?.Cancel();
+                Client?.Close();
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(ex.Message);
+            }
+        }
+
+        public void TX(string s)
```

Comment:
`TX` ir `RX` pavadinimai galetu buti aiskesni :D is pavadinimo nelabai aisku ka turetu daryti, ramybes negautum nuo kolegu jei iseitum atostogu tokius pavadinimus palikes :D